packages/main/src/FileUploader.ts

```diff
@@ -317,6 +367,32 @@ class FileUploader extends UI5Element implements IFormInputElement {
 		}, "");
 	}
 
+	/**
+	 * Checks whether all files are below `maxFileSize` (if set),
+	 * and fires a `fileSizeExceeded` event if any file exceeds it.
+	 * @private
+	 */
+	_validateFiles(changedFiles: FileList): FileList {
+		const exceededFilesData = this.maxFileSize ? this._getExceededFiles(changedFiles) : [];
+
+		if (exceededFilesData.length) {
+			this.fireEvent<FileUploaderFileSizeExceededEventDetail>("fileSizeExceeded", {
+				filesData: exceededFilesData,
+			});
+			changedFiles = new DataTransfer().files;
+		}
+
+		return changedFiles;
+	}
+
+	_getExceededFiles(files: FileList): FileData[] {
+		return Array.from(files)
```

Comment:
I think this can be done in one for loop. With filter and map we do 2(even 3 if we count the from method) iterations over the array, and if we have 1 for loop with check if it's larger and then put it in the array. We can even iterate it in for loop, so we will replace 3 iterations with 1.